File: server/pkg/model/session.go
package model

type SessionStatus int

const (
	SessionActive   SessionStatus = 0
	SessionInactive SessionStatus = 1
	SessionExpired  SessionStatus = 2
)

type DeviceType int

const (
	DevicePhone   DeviceType = 0
	DeviceDesktop DeviceType = 1
	DeviceWeb     DeviceType = 2
	DeviceTablet  DeviceType = 3
)

type Session struct {
	SessionID  string         `json:"session_id"`
	UserID     string         `json:"user_id"`
	Device     DeviceType     `json:"device"`
	DeviceName string         `json:"device_name"`
	DeviceID   string         `json:"device_id,omitempty"`
	ClientIP   string         `json:"client_ip,omitempty"`
	ConnID     string         `json:"conn_id,omitempty"`
	Status     SessionStatus  `json:"status"`
	LoginAt    int64          `json:"login_at"`
	LastActive int64          `json:"last_active"`
	Metadata   map[string]any `json:"metadata,omitempty"`
}

File: server/web/src/types/session.ts
export enum SessionStatus {
  Active = 0,
  Inactive = 1,
  Expired = 2,
}

export enum DeviceType {
  Phone = 0,
  Desktop = 1,
  Web = 2,
  Tablet = 3,
}

export interface Session {
  session_id: string
  user_id: string
  device: DeviceType
  device_name: string
  device_id?: string
  client_ip?: string
  conn_id?: string
  status: SessionStatus
  login_at: number
  last_active: number
  metadata?: Record<string, unknown>
}
